packages/panna-sdk/src/react/components/add-funds/transfer-from-wallet/transfer-amount-step.tsx

```diff
@@ -0,0 +1,442 @@
+import { ChevronDownIcon } from 'lucide-react';
+import { useEffect, useMemo, useState } from 'react';
+import { UseFormReturn } from 'react-hook-form';
+import { getWalletName, isWalletId, type WalletIdValue } from 'src/core';
+import { TokenBalance } from '@/mocks/token-balances';
+import { getEnvironmentChain, renderCryptoAmount } from '@/utils';
+import { truncateAddress } from '@/utils/address';
+import {
+  useAccountAvatar,
+  useActiveAccount,
+  useExternalWallet,
+  usePanna,
+  useTokenBalances
+} from '../../../hooks';
+import { Button } from '../../ui/button';
+import { Card } from '../../ui/card';
+import { DialogHeader, DialogTitle } from '../../ui/dialog';
+import { useDialogStepper } from '../../ui/dialog-stepper';
+import {
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage
+} from '../../ui/form';
+import { Input } from '../../ui/input';
+import { Label } from '../../ui/label';
+import { Separator } from '../../ui/separator';
+import { Typography } from '../../ui/typography';
+import { TransferFormData } from './schema';
+
+type TransferAmountStepProps = {
+  form: UseFormReturn<TransferFormData>;
+};
+
+// Amount display component for input endAdornment - shows USD and crypto equivalent
+function AmountDisplay({
+  tokenInfo,
+  cryptoAmount
+}: {
+  tokenInfo: TokenBalance;
+  cryptoAmount: string;
+}) {
+  // Format the crypto amount - remove trailing zeros
+  const formattedAmount = parseFloat(cryptoAmount)
+    .toFixed(6)
+    .replace(/\.?0+$/, '');
+
+  return (
+    <div className="flex flex-col pr-3 text-right">
+      <Typography variant="small" className="text-primary">
+        USD
+      </Typography>
+      <Typography variant="muted" className="text-xs">
+        {formattedAmount} {tokenInfo.token.symbol}
+      </Typography>
+    </div>
+  );
+}
+
+export function TransferAmountStep({ form }: TransferAmountStepProps) {
+  const { next } = useDialogStepper();
+  const account = useActiveAccount();
+  const { externalWallet } = useExternalWallet();
+  const { chainId } = usePanna();
+  const fromAddress = form.watch('fromAddress');
+  const tokenInfo = form.watch('tokenInfo') as TokenBalance;
+  const fiatAmount = form.watch('amount') || '0';
+
+  const sourceAddress =
+    fromAddress || externalWallet?.getAccount()?.address || '';
+
+  const chainInitial = useMemo(() => {
+    const chain = getEnvironmentChain(chainId);
+    const chainName = chain?.name ?? '';
+
+    if (chainName.length > 0) {
+      return chainName.charAt(0).toUpperCase();
+    }
+
+    return 'N';
+  }, [chainId]);
```

Comment:
Considering that we have images for Lisk and Ethereum in the `token-config.ts` and that most developers might be deploying on Lisk, we can actually show the chain image then fallback to initial. You can also leave it as is but it looks a bit confusing with the initial.